tests/test_queue.py

```diff
@@ -695,3 +698,505 @@ def test_job_in_failed_queue_persists(self):
         failed_queue.quarantine(job, Exception('Some fake error'))
 
         self.assertEqual(self.testconn.ttl(job.key), -1)
+
+class TestQueueCluster(RQTestClusterCase):
```

Comment:
We're duplicating a lot of tests. Is it possible to run all the tests twice, once with regular Redis connection and the second time with Redis cluster?

Reply:
I notice this too.My opition is that if rq supports redis cluster, all test cases should run with redis cluster to ensure rq support redis cluster well.By now, cluster mode is based on [redis-py-cluster](https://github.com/Grokzen/redis-py-cluster),redis-py-cluster does not support some operation like `watch` which rq needs. But do not mind that, I will work on redis-py-cluster to make it support operations rq needs.

By the way, I try to run test case once on both regular Redis connection and Redis cluster but failed because I am not familiar with rq code now.

Reply:
I can probably help with the RQ code base if we want to make it Redis cluster compatible. The changes should come in small PR batches though. I'm not in favor of a huge PR that changes a huge chunk of the code.

Reply:
Thanks @selwin .If RQ developers want it support Redis cluster as feature. I suggest you create a feature branch. I will make PR batches to the branch.

Reply:
Could you contribute the patches directly to the master branch? This way, we wouldn't need to worry about keeping the feature branch in sync with the master.

I'm personally getting a new PR ready that reworks how `FailedQueue` works, in preparation for the 1.0 release.

Reply:
OK, I am working on https://github.com/Grokzen/redis-py-cluster/pull/251. It will take some time. RQ code needs `watch` method which uses Redis pipeline and transaction feature. So I will test this carefully in redis cluster mode.